homeassistant/components/group/media_player.py

```diff
@@ -0,0 +1,350 @@
+        )
+
+    def media_seek(self, position):
+        """Send seek command."""
+        data = {
+            ATTR_MEDIA_SEEK_POSITION: position,
+            ATTR_ENTITY_ID: self._players[KEY_SEEK],
+        }
+        self.hass.services.call(
+            DOMAIN, SERVICE_MEDIA_SEEK, data, blocking=True, context=self._context,
+        )
+
+    def media_stop(self):
+        """Send stop command."""
+        data = {ATTR_ENTITY_ID: self._players[KEY_PAUSE_PLAY_STOP]}
+        self.hass.services.call(
+            DOMAIN, SERVICE_MEDIA_STOP, data, blocking=True, context=self._context,
+        )
+
+    def mute_volume(self, mute):
+        """Mute the volume."""
+        data = {
+            ATTR_MEDIA_VOLUME_MUTED: mute,
+            ATTR_ENTITY_ID: self._players[KEY_VOLUME],
+        }
+        self.hass.services.call(
+            DOMAIN, SERVICE_VOLUME_MUTE, data, blocking=True, context=self._context,
+        )
+
+    def play_media(self, media_type, media_id, **kwargs):
+        """Play a piece of media."""
+        data = {
+            ATTR_MEDIA_CONTENT_ID: media_id,
+            ATTR_MEDIA_CONTENT_TYPE: media_type,
+            ATTR_ENTITY_ID: self._players[KEY_PLAY_MEDIA],
+        }
+        self.hass.services.call(
+            DOMAIN, SERVICE_PLAY_MEDIA, data, blocking=True, context=self._context,
+        )
+
+    def turn_on(self):
+        """Forward the turn_on command to all media in the media group."""
+        data = {ATTR_ENTITY_ID: self._players[KEY_ON_OFF]}
+        self.hass.services.call(
+            DOMAIN, SERVICE_TURN_ON, data, blocking=True, context=self._context,
+        )
+
+    def set_volume_level(self, volume):
+        """Set volume level(s)."""
+        data = {
+            ATTR_MEDIA_VOLUME_LEVEL: volume,
+            ATTR_ENTITY_ID: self._players[KEY_VOLUME],
+        }
+        self.hass.services.call(
+            DOMAIN, SERVICE_VOLUME_SET, data, blocking=True, context=self._context,
+        )
+
+    def turn_off(self):
+        """Forward the turn_off command to all media in the media group."""
+        data = {ATTR_ENTITY_ID: self._players[KEY_ON_OFF]}
+        self.hass.services.call(
+            DOMAIN, SERVICE_TURN_OFF, data, blocking=True, context=self._context,
+        )
+
+    async def async_volume_up(self):
+        """Turn volume up for media player(s)."""
+        for entity in self._players[KEY_VOLUME]:
+            volume_level = self.hass.states.get(entity).attributes["volume_level"]
+            if volume_level < 1:
+                await self.async_set_volume_level(min(1, volume_level + 0.1))
+
+    async def async_volume_down(self):
+        """Turn volume down for media player(s)."""
+        for entity in self._players[KEY_VOLUME]:
+            volume_level = self.hass.states.get(entity).attributes["volume_level"]
+            if volume_level > 0:
+                await self.async_set_volume_level(max(0, volume_level - 0.1))
+
+    async def async_update(self):
```

Comment:
Since we don't use the default pooling mechanism we can rename this method and use `self.async_write_ha_state` here to update entity state instead of calling `self.async_schedule_update_ha_state(True)`.

Reply:
Done